pyramid_tm/__init__.py

```diff
@@ -1,15 +1,21 @@
 import sys
 import transaction
 
+from pyramid.httpexceptions import HTTPNotFound
+from pyramid.interfaces import IRequestExtensions
+from pyramid.interfaces import IRequestFactory
+from pyramid.request import Request, apply_request_extensions
 from pyramid.settings import asbool
-from pyramid.util import DottedNameResolver
+from pyramid.threadlocal import manager as request_manager
```

Comment:
It's not unprecedented for official pyramid addons to use private pyramid apis, despite being ugly and unfortunate... it's necessary here until pyramid grows some other way to copy the request.